sql/parser/eval.go

```diff
@@ -547,10 +540,12 @@ type EvalContext struct {
 	NodeID        uint32
 	StmtTimestamp DTimestamp
 	TxnTimestamp  DTimestamp
+	ReCache       *RegexpCache
 	GetLocation   func() (*time.Location, error)
 }
 
 var defaultContext = EvalContext{
+	ReCache: NewRegexpCache(128), // TODO(nvanbenschoten) how should this be sized?
```

Comment:
I think I'd just leave this nil and handling the nil receiver case in the methods on RegexpCache; `defaultContext` should never need to use the cache.